rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2217,19 +2238,23 @@ void rskTxWaitingForSignature_fail_adding_an_already_existence_key_after_rskip_3
             .build();
         bridgeSupport.updateCollections(tx);
 
-        // Since only one pegout per updateCollection call is move to waitingForisgnatures
-        // -assert one of the two pegout get moved to pegoutWaitingForSignate and one is left on waitingForConfirmation
-        assertEquals(1, provider.getPegoutsWaitingForConfirmations().getEntries().size());
-        assertEquals(1, provider.getPegoutsWaitingForSignatures().size());
+        /*
+            Since only one pegout per updateCollection call is move to waitingForSignatures
+            assert one of the two pegout get moved to pegoutWaitingForSignate and one is left on waitingForConfirmation
+         */
+        assertEquals(1, pegoutsWaitingForConfirmations.getEntries().size());
+        assertEquals(1, pegoutWaitingForSignatures.size());
+
+        // assert the migration tx was the first one to be confirmed
+        assertTrue(pegoutWaitingForSignatures.containsKey(migrationTx.getPegoutCreationRskTxHash()));
 
         // Advance blockchain one block to make the second pegout to be moved to waitingForConfirmation
```

Comment:
This isn't really what's happening. We advance the blockchain but no transaction is moved because the key is duplicated

Reply:
No, this is the first updateCollection call. It's expected that one tx gets moved to pegoutWaitingForSignatures. They can be duplicated in the `pegoutWaitingForConfirmation` and one of them be moved to `pegoutWaitingForSignatures`, there is no error in doing that.